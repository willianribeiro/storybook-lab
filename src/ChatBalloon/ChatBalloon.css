.chat-balloon {
    padding: 24px;
    position: relative;
    border-radius: 5px;
    background-color: #fff;
    box-shadow:
        0 1px 3px rgba(0, 0, 0, 0.12),
        0 1px 2px rgba(0, 0, 0, 0.24);
    font-size: 14px;
}

.chat-balloon .chat-balloon__header {
    display: flex;
    align-items: center;
    padding-bottom: 24px;
    font-size: 16px;
    font-weight: bold;
    line-height: 1.5;
}

.chat-balloon .chat-balloon__body {
    font-size: 14px;
    line-height: 1.5;
}

/* CARET */
.chat-balloon::before,
.chat-balloon::after {
    content: '';
    position: absolute;
    border-width: 10px 15px;
    border-style: solid;
    border-color: transparent;
}

/* CARET - BACKGROUND */
.chat-balloon::before {
    top: 32px;
}

/* CARET - FOREGROUND */
.chat-balloon::after {
    top: 30px;
}


/* STATES - CHAT BALLOON */
.chat-balloon.is-caret-left::before,
.chat-balloon.is-caret-left::after {
    left: -30px;
}

.chat-balloon.is-caret-left::before {
    border-right-color: rgba(0, 0, 0, .24);
}

.chat-balloon.is-caret-left::after {
    border-right-color:#fff;
}

.chat-balloon.is-caret-left.is-dark {
    background-color: #f4f4f4;
}

.chat-balloon.is-caret-left.is-darkbefore {
    border-right-color: rgba(0, 0, 0, .24);
}

.chat-balloon.is-caret-left.is-darkafter {
    border-right-color:#f4f4f4;
}

.chat-balloon.is-caret-right::before,
.chat-balloon.is-caret-right::after {
    right: -30px;
}

.chat-balloon.is-caret-right::before {
    border-left-color: rgba(0, 0, 0, .24);
}

.chat-balloon.is-caret-right::after {
    border-left-color:#fff;
}

.chat-balloon.is-caret-right.is-dark {
    background-color: #f4f4f4;
}

.chat-balloon.is-caret-right.is-dark::before {
    border-left-color: rgba(0, 0, 0, .24);
}

.chat-balloon.is-caret-right.is-dark::after {
    border-left-color:#f4f4f4;
}

