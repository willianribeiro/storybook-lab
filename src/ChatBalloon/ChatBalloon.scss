@import '../core/sass/main';

// CHAT BALLOON
.chat-balloon {
    padding: $SPACING_3;
    position: relative;
    border-radius: $BORDER_RADIUS;
    background-color: $COLOR_WHITE;
    @include m-box-shadow(1);
    font-size: 14px;

    .chat-balloon__header {
        display: flex;
        align-items: center;
        padding-bottom: $SPACING_3;
        font-size: 16px;
        font-weight: bold;
        line-height: 1.5;
    }

    .chat-balloon__body {
        font-size: 14px;
        line-height: 1.5;
    }

    // CARET
    &::before,
    &::after {
        content: '';
        position: absolute;
        border-width: 10px 15px;
        border-style: solid;
        border-color: transparent;
    }

    // CARET - BACKGROUND
    &::before {
        top: 32px;
    }

    // CARET - FOREGROUND
    &::after {
        top: 30px;
    }


    // STATES - CHAT BALLOON
    &.is-caret-left {
        &::before,
        &::after {
            left: -30px;
        }

        &::before {
            border-right-color: rgba(0, 0, 0, .24);
        }

        &::after {
            border-right-color:$COLOR_WHITE;
        }
    }

    &.is-caret-left.is-dark {
        background-color: $COLOR_GREY_LIGHTER;

        &::before {
            border-right-color: rgba(0, 0, 0, .24);
        }

        &::after {
            border-right-color:$COLOR_GREY_LIGHTER;
        }
    }

    &.is-caret-right {
        &::before,
        &::after {
            right: -30px;
        }

        &::before {
            border-left-color: rgba(0, 0, 0, .24);
        }

        &::after {
            border-left-color:$COLOR_WHITE;
        }
    }

    &.is-caret-right.is-dark {
        background-color: $COLOR_GREY_LIGHTER;

        &::before {
            border-left-color: rgba(0, 0, 0, .24);
        }

        &::after {
            border-left-color:$COLOR_GREY_LIGHTER;
        }
    }
}
