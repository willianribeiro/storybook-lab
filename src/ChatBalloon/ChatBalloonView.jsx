import React, { PropTypes } from 'react';

const ChatBalloonView = ({
  headerContent,
  bodyContent,
  className,
  }) => (
    <div className={className}>
      <div className="chat-balloon__header">
        { headerContent }
      </div>

      <div className="chat-balloon__body">
        { bodyContent }
      </div>
    </div>
);

ChatBalloonView.propTypes = {
  headerContent: PropTypes.node.isRequired,
  bodyContent: PropTypes.node.isRequired,
  className: PropTypes.string.isRequired,
};

export default ChatBalloonView;
