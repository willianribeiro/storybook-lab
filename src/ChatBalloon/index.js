export { default as ChatBalloon } from './ChatBalloon';
