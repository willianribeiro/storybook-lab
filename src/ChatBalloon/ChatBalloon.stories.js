import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import ChatBalloon from './ChatBalloon';


storiesOf('ChatBalloon', module)
  .addDecorator((story) => (
    <div>
      <h1>ChatBalloon</h1>
      <div> { story() } </div>
    </div>
  ))
  .add('Simplest', () => (
    <ChatBalloon
      headerContent='ChatBallon Simples'
      bodyContent='Mussum Ipsum, cacilds vidis litro abertis. Leite de capivaris, leite de mula manquis sem cabeça. Interagi no mé, cursus quis, vehicula ac nisi. Manduma pindureta quium dia nois paga. In elementis mé pra quem é amistosis quis leo.'
    />
  ))
  .add('Dark theme', () => (
    <ChatBalloon
      darkTheme
      headerContent='Com tema Dark'
      bodyContent='Mussum Ipsum, cacilds vidis litro abertis. Leite de capivaris, leite de mula manquis sem cabeça. Interagi no mé, cursus quis, vehicula ac nisi. Manduma pindureta quium dia nois paga. In elementis mé pra quem é amistosis quis leo.'
    />
  ))
  .add('Caret on the right', () => (
    <ChatBalloon
      caretRight
      headerContent='Com caret na direita'
      bodyContent='Mussum Ipsum, cacilds vidis litro abertis. Leite de capivaris, leite de mula manquis sem cabeça. Interagi no mé, cursus quis, vehicula ac nisi. Manduma pindureta quium dia nois paga. In elementis mé pra quem é amistosis quis leo.'
    />
  ))
  .add('Without caret', () => (
    <ChatBalloon
      removeCaret
      headerContent='Sem caret'
      bodyContent='Mussum Ipsum, cacilds vidis litro abertis. Leite de capivaris, leite de mula manquis sem cabeça. Interagi no mé, cursus quis, vehicula ac nisi. Manduma pindureta quium dia nois paga. In elementis mé pra quem é amistosis quis leo.'
    />
  ))
  .add('With HTML on the body', () => {
    const bodyHTML = (
      <div>
        Mussum Ipsum, cacilds vidis <strong>litro abertis. </strong>
        Leite de capivaris, leite de mula manquis sem cabeça.
        <br />
        <br />
        <img
          src='http://bit.ly/1PPeefG'
          alt="pug"
          style={{ maxWidth: '100%'}}
        />
      </div>
    );

    return (
      <ChatBalloon
        headerContent='Passando um HTML como parâmetro para o body'
        bodyContent={bodyHTML}
      />
    );
  });
