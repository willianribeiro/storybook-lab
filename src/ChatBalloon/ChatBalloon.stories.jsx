import React from 'react';
import { storiesOf } from '@storybook/react';

import ChatBalloon from './ChatBalloon';

let bodyContent = '';
bodyContent += 'Mussum Ipsum, cacilds vidis litro abertis. Leite de capivaris, leite de mula ';
bodyContent += 'manquis sem cabeça. Interagi no mé, cursus quis, vehicula ac nisi. ';
bodyContent += 'Manduma pindureta quium dia nois paga. In elementis mé pra quem ';
bodyContent += 'é amistosis quis leo.';

storiesOf('ChatBalloon', module)
  .addDecorator(story => (
    <div>
      <h1 className="story-title">ChatBalloon</h1>
      <div> { story() } </div>
    </div>
  ))
  .addWithInfo('Simplest', () => (
    <ChatBalloon
      headerContent="ChatBalloon Simples"
      bodyContent={bodyContent}
    />
  ))
  .addWithInfo('Dark theme', () => (
    <ChatBalloon
      darkTheme
      headerContent="Com tema Dark"
      bodyContent={bodyContent}
    />
  ))
  .addWithInfo('Caret on the right', () => (
    <ChatBalloon
      caretRight
      headerContent="Com caret na direita"
      bodyContent={bodyContent}
    />
  ))
  .addWithInfo('Without caret', () => (
    <ChatBalloon
      removeCaret
      headerContent="Sem caret"
      bodyContent={bodyContent}
    />
  ))
  .addWithInfo('With HTML on the body', () => {
    const bodyHTML = (
      <div>
        Mussum Ipsum, cacilds vidis <strong>litro abertis. </strong>
        Leite de capivaris, leite de mula manquis sem cabeça.
        <br />
        <br />
        <img
          src="http://i.imgur.com/qpdJh22.jpg"
          alt="pug"
          style={{ maxWidth: '100%' }}
        />
      </div>
    );

    return (
      <ChatBalloon
        headerContent="Passando um HTML como parâmetro para o body"
        bodyContent={bodyHTML}
      />
    );
  });
