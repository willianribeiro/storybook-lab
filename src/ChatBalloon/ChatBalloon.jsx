import React from 'react';
import PropTypes from 'prop-types';

import classnames from 'classnames';

import ChatBalloonView from './ChatBalloonView';

const ChatBalloon = ({
  headerContent,
  bodyContent,
  removeCaret,
  caretRight,
  darkTheme,
  className }) => {
  const classNames = classnames(
    className,
    'chat-balloon',
    { 'is-caret-right': caretRight && !removeCaret },
    { 'is-caret-left': !caretRight && !removeCaret },
    { 'is-dark': darkTheme },
  );

  return (
    <ChatBalloonView
      headerContent={headerContent}
      bodyContent={bodyContent}
      removeCaret={removeCaret}
      caretRight={caretRight}
      className={classNames}
    />
  );
};

ChatBalloon.propTypes = {
  headerContent: PropTypes.node.isRequired,
  bodyContent: PropTypes.node.isRequired,
  removeCaret: PropTypes.bool,
  caretRight: PropTypes.bool,
  darkTheme: PropTypes.bool,
  className: PropTypes.string,
};

ChatBalloon.defaultProps = {
  removeCaret: false,
  caretRight: false,
  darkTheme: false,
  className: '',
};

export default ChatBalloon;
