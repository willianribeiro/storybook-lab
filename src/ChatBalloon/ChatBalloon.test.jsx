import React from 'react';
import renderer from 'react-test-renderer';

import ChatBalloon from './ChatBalloon';

it('should render ChatBalloon correctly', () => {
  let bodyContent = '';
  bodyContent += 'Mussum Ipsum, cacilds vidis litro abertis. Leite de capivaris, leite de mula ';
  bodyContent += 'manquis sem cabeça. Interagi no mé, cursus quis, vehicula ac nisi. ';
  bodyContent += 'Manduma pindureta quium dia nois paga. In elementis mé pra quem ';
  bodyContent += 'é amistosis quis leo.';

  const tree = renderer.create(
    <ChatBalloon
      headerContent="Chat Balloon Simples"
      bodyContent={bodyContent}
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
