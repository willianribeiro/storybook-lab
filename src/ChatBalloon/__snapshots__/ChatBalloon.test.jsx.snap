// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ChatBalloon correctly 1`] = `
<div
  className="chat-balloon is-caret-left"
>
  <div
    className="chat-balloon__header"
  >
    Chat Balloon Simples
  </div>
  <div
    className="chat-balloon__body"
  >
    Mussum Ipsum, cacilds vidis litro abertis. Leite de capivaris, leite de mula manquis sem cabeça. Interagi no mé, cursus quis, vehicula ac nisi. Manduma pindureta quium dia nois paga. In elementis mé pra quem é amistosis quis leo.
  </div>
</div>
`;
